tests/components/stiebel_eltron/test_climate.py

```diff
@@ -0,0 +1,312 @@
+"""Test the STIEBEL ELTRON climate entity."""
+
+from unittest.mock import MagicMock
+
+from pymodbus.exceptions import ModbusException
+from pystiebeleltron.lwz import OperatingMode
+import pytest
+
+from homeassistant.components.climate import (
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_PRESET_MODES,
+    ATTR_TARGET_TEMP_STEP,
+    DATA_COMPONENT,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.stiebel_eltron.climate import (
+    HA_TO_LWZ_HVAC,
+    HA_TO_LWZ_PRESET,
+    LWZ_TO_HA_HVAC,
+    LWZ_TO_HA_PRESET,
+    PRESET_AUTO,
+    PRESET_EMERGENCY,
+    PRESET_MANUAL,
+    PRESET_READY,
+    PRESET_WATER_HEATING,
+)
+from homeassistant.const import (
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+    STATE_UNKNOWN,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+from tests.components.climate.common import (
+    async_set_hvac_mode,
+    async_set_preset_mode,
+    async_set_temperature,
+)
+
+CLIMATE_UNIQUE_ID = "stiebel_eltron-lwz-climate_hk_1"
+
+
+async def _create_climate_entry(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    config_entry: MockConfigEntry,
+) -> er.RegistryEntry:
+    """Set up the integration and return the climate registry entry."""
+
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    for registry_entry in er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
+    ):
+        if registry_entry.unique_id == CLIMATE_UNIQUE_ID:
+            return registry_entry
+    pytest.fail("Expected climate registry entry not found")
+
+
+async def test_climate_entity_setup(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test the climate entity is set up correctly via Home Assistant."""
+
+    # Create the entity and refresh its state
+    climate_entry = await _create_climate_entry(
+        hass, entity_registry, mock_config_entry
+    )
+    component = hass.data[DATA_COMPONENT]
+    entity = component.get_entity(climate_entry.entity_id)
+
+    # Test entity properties
+    assert entity is not None
+    assert entity.unique_id == CLIMATE_UNIQUE_ID
+    assert entity.temperature_unit == UnitOfTemperature.CELSIUS
```

Comment:
Okay this is already way better than before. Except, we should not touch entities directly. Instead we should interact with them via the state machine.

So now indeed the climate entity is created, now you can get everything you need via `hass.states.get("climate.something")` and assert that way. Below you have a lot of assertions, have a look at `snapshot_platform` helper, as that will generate `.ambr` files that fixate both the state and the entity registry entry. It can be generated using `pytest ./tests/components/stiebel_eltron --snapshot-update`